src/cpp/shared-table.cpp

```diff
@@ -280,7 +280,8 @@ SharedTable SharedTable::luaSetMetatable(const sol::stack_object& tbl, const sol
         stable.ctx_->metatable = GCNull;
     }
 
-    stable.ctx_->metatable = createStoredObject(mt)->gcHandle();
+    const auto mtObj = createStoredObject(mt);
```

Comment:
Where is the race?

Reply:
returned stored object could be destroyed by GC before we add using `addReference`

Reply:
returned stored object could be destroyed by GC before we add using `addReference`

Reply:
returned stored object could be destroyed by GC before we add using `addReference`

Reply:
returned stored object could be destroyed by GC before we add using `addReference`

Reply:
returned stored object could be destroyed by GC before we add using `addReference`

Reply:
returned stored object could be destroyed by GC before we add it using `addReference`

Reply:
returned stored object could be destroyed by GC before we add it using `addReference`